junebug/api.py

```diff
@@ -699,9 +699,15 @@ def return_queue_results(results):
                             'name': queue['name'],
                             'stuck': False,
                             'messages': queue.get('messages'),
-                            'rate':
-                                queue['message_stats']['ack_details']['rate']
                         }
+
+                        # Use the rate from message_stats if it is there
+                        if("message_stats" in queue):
+                            details['rate'] = \
+                                queue['message_stats']['ack_details']['rate']
+                        else:
+                            details['rate'] = queue['messages_details']['rate']
```

Comment:
I don't think that we should be using `messages_details` rate at all, it's very misleading.

I'm taking a wild guess that if a queue hasn't seen activity for a long while (quite possible in QA), then it won't have the `message_stats` key.

If that's the case, I would rather skip checking that queue than rely on `messages_details.rate`.